### PR TITLE
Restuctured tests

### PR DESCRIPTION
**Pull request by ubuntolog**
I put all the tests into a separate package

### CODE REVIEW

**Comment by ubuntolog** [comment]
Build failed

File: backend-docker/main.go [hunk 1]
```diff
@@ -1,58 +1,40 @@
...
 )
 
-var configFilePath = "config.json"
-var config configuration
-
-type configuration struct {
-	Docker []dckr.Config
-	Server server.Config
-}
+var configFilePath = "config/config.json"
```
Comment (emanueldima) [inline backend-docker/main.go]: I disagree with this change, the config file must be placed in the base directory, to be easily found when we run the compiled binary.

File: backend-docker/main.go [hunk 2]
```diff
@@ -1,58 +1,40 @@
...
 
 func main() {
 	flag.StringVar(&configFilePath, "config", configFilePath, "configuration file")
 	flag.Parse()
 
-	var err error
-	config, err = readConfigFile(configFilePath)
+	//var err error
```
Comment (emanueldima) [inline backend-docker/main.go]: remove this line

File: backend-docker/tests/apiserver_test.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	s := server.NewServer(settings.Server, c)
+
+	srv := httptest.NewServer(s.Server.Handler)
+	baseURL := srv.URL + "/api/"
+
+	ifAPIExist(baseURL, t)
+	callListVolumesHandler(baseURL + "volumes", t)
+}
+
+func ifAPIExist(callURL string, t *testing.T) bool {
```
Comment (emanueldima) [inline backend-docker/tests/apiserver_test.go]: Recommendation: it's better to start function names with a verb. I would name this `checkIfAPIExists`

File: backend-docker/tests/apiserver_test.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			k[i] = s
+			i++
+		}
+		log.Println(k[0])
+
+		// the only found key should look like this
+		if k[0] != "Volumes" {
+			t.Error("Reply is not correct")
+			t.Fail()
+		}
```
Comment (emanueldima) [inline backend-docker/tests/apiserver_test.go]: I think you can make the same test in a simpler manner: declare a local struct with a Volumes field and check that Unmarshalling into that structure works. Something like this:

````
struct VolumeField { Volumes: map[string]interface{} }
var volumeField VolumeField
err = json.Unmarshal(htmlData, &volumeField)
if err != nil { 
    t.Error() 
}
````

Comment (ubuntolog) [inline backend-docker/tests/apiserver_test.go]: Yes, it looks much nicer and I tried a similar thing but I got "json: cannot unmarshal array into Go value of type map[string]interface {}"

Comment (emanueldima) [inline backend-docker/tests/apiserver_test.go]: Have you tried with a simple array, then: `struct VolumeField { Volumes: []interface{} }` ?

Comment (ubuntolog) [inline backend-docker/tests/apiserver_test.go]: No

Comment (ubuntolog) [inline backend-docker/tests/apiserver_test.go]: Ok, it passes the test. But it also passes the test when we have something else instead of "Volumes"

**Pull request merged**